mvlearn/embed/kcca.py

```diff
@@ -3,29 +3,58 @@
 ====================================
 Python module for kernel canonical correlation analysis (kCCA)
 
-Code modified from UC Berkeley, Gallant lab
-(https://github.com/gallantlab/pyrcca)
-Copyright 2016, UC Berkeley, Gallant lab.
+Adopted from Steven Van Vaerenbergh's MATLAB Package 'KMBOX'
+https://github.com/steven2358/kmbox
 """
 
 from .base import BaseEmbed
 from ..utils.utils import check_Xs
 
 import numpy as np
-from scipy.linalg import eigh
+import numpy.matlib
+from scipy import linalg
+from sklearn.metrics.pairwise import euclidean_distances
 
 
 class KCCA(BaseEmbed):
-    """
-    CCA aims to find useful projections of the
+    r"""
+    The kernel canonical correlation analysis (KCCA) is a method
+    that generalizes the classical linear canonical correlation
+    analysis (CCA) to nonlinear setting.  It allows us to depict the
+    nonlinear relation of two sets of variables and enables
+    applications of classical multivariate data analysis
+    originally constrained to linearity relation (CCA).
+
+    Parameters
+    ----------
+    reg : float, default = 0.00001
+          Regularization parameter
+    n_components : int, default = 10
+                   Number of canonical dimensions to keep
```

Comment:
can you make n_components the first argument, rather than reg? Maybe push reg towards the end of the list, since it seems much less important

Reply:
completed